### PR TITLE
ADM-471: [frontend] Error handling enhancement for frontend 

### PR DESCRIPTION
**Pull request by TingLiMio**
## Summary

- Use hintInfo defined in backend to show error message.
- Handle Timeout exception.

### CODE REVIEW

File: frontend/__tests__/src/client/CSVClient.test.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { setupServer } from 'msw/node'
 import { rest } from 'msw'
 import { HttpStatusCode } from 'axios'
-import { MOCK_EXPORT_CSV_REQUEST_PARAMS, MOCK_EXPORT_CSV_URL, VERIFY_ERROR_MESSAGE } from '../fixtures'
+import { MOCK_EXPORT_CSV_REQUEST_PARAMS, MOCK_EXPORT_CSV_URL } from '../fixtures'
 import { csvClient } from '@src/clients/report/CSVClient'
+import * as http from 'http'
```
Comment (guzhongren) [inline frontend/__tests__/src/client/CSVClient.test.ts:6]: Try to ‘import’ all deps explicitly,not use ‘*’

**Review by guzhongren — COMMENTED** [review]
Need some change with the comments

File: frontend/__tests__/src/client/BoardClient.test.ts [hunk 2]
```diff
@@ -59,19 +67,44 @@ describe('verify board request', () => {
...
+
   it('should throw error when board verify response status 300', async () => {
     server.use(rest.get(MOCK_BOARD_URL_FOR_JIRA, (req, res, ctx) => res(ctx.status(HttpStatusCode.MultipleChoices))))
 
     await expect(async () => {
       await boardClient.getVerifyBoard(MOCK_BOARD_VERIFY_REQUEST_PARAMS)
     }).rejects.toThrow(VERIFY_ERROR_MESSAGE.UNKNOWN)
   })
+
+  it('should throw unknown error when board verify response empty', async () => {
```
[the remark this answers is not in the document]
Comment (TingLiMio) [reply frontend/__tests__/src/client/BoardClient.test.ts:103]: No Content status code is `204`, the data is empty. Currently tests the entire response is empty, including status code, data, etc.

File: frontend/__tests__/src/client/CSVClient.test.ts (hunk 1 above)
Comment (TingLiMio) [reply frontend/__tests__/src/client/CSVClient.test.ts]: Updated!

**Review by guzhongren — COMMENTED** [review]
LGTM

**Review by ziwen-xiao — APPROVED** [review]

**Pull request merged**